pkg/labels/labels.go

```diff
@@ -61,6 +61,13 @@ func (ls Set) AsSelector() Selector {
 	return SelectorFromSet(ls)
 }
 
+// ValidatedAsSelector converts labels into a selector, but
+// assumes that labels are already validated and thus don't
+// preform any validation.
```

Comment:
"we measured and this is significantly faster in codepaths that matter at high scale"